From: Markus S. <markus.s@kdemail.net>
From: Will Stephenson <wstephenson@kde.org>
Subject: AOL sold ICQ. Change login.oscar.aol.com to login.icq.com
Bug: https://bugs.kde.org/show_bug.cgi?id=257008
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=603661
Origin: backport
 svn diff svn://anonsvn.kde.org/home/kde -c 1198791
 svn diff svn://anonsvn.kde.org/home/kde -c 1197799
Forwarded: yes
Last-Update: 2010-11-28

Change login.oscar.aol.com to login.icq.com and update existing ICQ accounts
using the mechanism provided by kdelibs.

--- a/kopete/kopete/kconf_update/CMakeLists.txt
+++ b/kopete/kopete/kconf_update/CMakeLists.txt
@@ -4,9 +4,9 @@ if(NOT WIN32)
 endif(NOT WIN32)
 
 
-install( FILES kopete-pluginloader.upd  kopete-nameTracking.upd  kopete-initialstatus.upd  kopete-gaim_to_pidgin_style.upd kopete-update_yahoo_server.upd  DESTINATION ${KCONF_UPDATE_INSTALL_DIR})
+install( FILES kopete-pluginloader.upd  kopete-nameTracking.upd  kopete-initialstatus.upd  kopete-gaim_to_pidgin_style.upd kopete-update_yahoo_server.upd kopete-update_icq_server.upd DESTINATION ${KCONF_UPDATE_INSTALL_DIR})
 
-install( PROGRAMS kopete-pluginloader.pl kopete-update_yahoo_server.pl
+install( PROGRAMS kopete-pluginloader.pl kopete-update_yahoo_server.pl kopete-update_icq_server.pl
 	kopete-account-0.10.pl kopete-initialstatus.pl kopete-gaim_to_pidgin_style.pl
   DESTINATION ${KCONF_UPDATE_INSTALL_DIR})
 
--- /dev/null
+++ b/kopete/kopete/kconf_update/kopete-update_icq_server.pl
@@ -0,0 +1,20 @@
+#!/usr/bin/perl
+
+# AOL sold ICQ; change the default server to login.icq.com
+
+my $inICQ = 0;
+foreach (<>) {
+    $inICQ = 1 if (/^\[Account_ICQProtocol_.*$/);
+    if ($inICQ) {
+        if (/^Server\=(.*)/) {
+            my $oldServer = $1;
+            if ($oldServer =~ m/\.aol\.com$/) {
+                print "Server=login.icq.com\n";
+                $inICQ = 0;
+                next;
+            }
+        }
+    }
+
+    print $_;
+}
--- /dev/null
+++ b/kopete/kopete/kconf_update/kopete-update_icq_server.upd
@@ -0,0 +1,4 @@
+Id=kopete-update-icq-server/r1
+File=kopeterc
+Options=overwrite
+Script=kopete-update_icq_server.pl,perl
--- a/kopete/protocols/oscar/icq/icqaccount.cpp
+++ b/kopete/protocols/oscar/icq/icqaccount.cpp
@@ -136,7 +136,7 @@ ICQAccount::ICQAccount(Kopete::Protocol
 	// Create actions
 	mEditInfoAction = new KAction( KIcon("user-properties"), i18n( "Edit User Info..." ), this );
 	QObject::connect( mEditInfoAction, SIGNAL(triggered(bool)), this, SLOT(slotUserInfo()) );
-	
+
 	mActionInvisible = new KToggleAction( i18n( "In&visible" ), this );
 	QObject::connect( mActionInvisible, SIGNAL(triggered(bool)), this, SLOT(slotToggleInvisible()) );
 
@@ -188,7 +188,7 @@ void ICQAccount::fillActionMenu( KAction
 	*/
 
 	KActionMenu *xtrazStatusMenu = new KActionMenu( i18n( "Set Xtraz Status" ), actionMenu );
-	
+
 	KAction* xtrazStatusSetAction = new KAction( i18n( "Set Status..." ), xtrazStatusMenu );
 	QObject::connect( xtrazStatusSetAction, SIGNAL(triggered(bool)), this, SLOT(setXtrazStatus()) );
 	xtrazStatusMenu->addAction( xtrazStatusSetAction );
@@ -236,7 +236,7 @@ void ICQAccount::connectWithPassword( co
 		myself()->setOnlineStatus( protocol()->statusManager()->connectingStatus() );
 		QString icqNumber = accountId();
 		kDebug(14153) << "Logging in as " << icqNumber;
-		QString server = configGroup()->readEntry( "Server", QString::fromLatin1( "login.oscar.aol.com" ) );
+		QString server = configGroup()->readEntry( "Server", QString::fromLatin1( "login.icq.com" ) );
 		uint port = configGroup()->readEntry( "Port", 5190 );
 
 		//set up the settings for the account
--- a/kopete/protocols/oscar/icq/ui/icqeditaccountwidget.cpp
+++ b/kopete/protocols/oscar/icq/ui/icqeditaccountwidget.cpp
@@ -65,7 +65,7 @@ ICQEditAccountWidget::ICQEditAccountWidg
 
 	mProtocol->fillComboFromTable( mAccountSettings->encodingCombo, mProtocol->encodings() );
 
-	//Setup the edtAccountId 
+	//Setup the edtAccountId
 	QRegExp rx("[0-9]{9}");
 	QValidator* validator = new QRegExpValidator( rx, this );
 	mAccountSettings->edtAccountId->setValidator(validator);
@@ -80,9 +80,9 @@ ICQEditAccountWidget::ICQEditAccountWidg
 		mAccountSettings->mPasswordWidget->load(&mAccount->password());
 		mAccountSettings->chkAutoLogin->setChecked(mAccount->excludeConnect());
 
-		QString serverEntry = mAccount->configGroup()->readEntry("Server", "login.oscar.aol.com");
+		QString serverEntry = mAccount->configGroup()->readEntry("Server", "login.icq.com");
 		int portEntry = mAccount->configGroup()->readEntry("Port", 5190);
-		if ( serverEntry != "login.oscar.aol.com" || ( portEntry != 5190) )
+		if ( serverEntry != "login.icq.com" || ( portEntry != 5190) )
 			mAccountSettings->optionOverrideServer->setChecked( true );
 
 		mAccountSettings->edtServerAddress->setText( serverEntry );
@@ -174,7 +174,7 @@ ICQEditAccountWidget::ICQEditAccountWidg
 			default:
 				encodingId=4;
 		}
-		
+
 		mProtocol->setComboFromTable( mAccountSettings->encodingCombo,
 		                              mProtocol->encodings(),
 		                              encodingId );
@@ -203,13 +203,13 @@ ICQEditAccountWidget::~ICQEditAccountWid
 {
 	if ( m_visibleEngine )
 		delete m_visibleEngine;
-	
+
 	if ( m_invisibleEngine )
 		delete m_invisibleEngine;
-	
+
 	if ( m_ignoreEngine )
 		delete m_ignoreEngine;
-	
+
 	delete mAccountSettings;
 }
 
@@ -254,7 +254,7 @@ Kopete::Account *ICQEditAccountWidget::a
 	}
 	else
 	{
-		mAccount->setServerAddress("login.oscar.aol.com");
+		mAccount->setServerAddress("login.icq.com");
 		mAccount->setServerPort(5190);
 	}
 
@@ -279,17 +279,17 @@ Kopete::Account *ICQEditAccountWidget::a
 	{
 		if ( m_visibleEngine )
 			m_visibleEngine->storeChanges();
-		
+
 		if ( m_invisibleEngine )
 			m_invisibleEngine->storeChanges();
-		
+
 		if ( m_ignoreEngine )
 			m_ignoreEngine->storeChanges();
 
 		//Update Oscar settings
 		static_cast<ICQMyselfContact*>( mAccount->myself() )->fetchShortInfo();
 	}
-	
+
 	return mAccount;
 }
 
@@ -299,7 +299,7 @@ bool ICQEditAccountWidget::validateData(
 	bool bOk;
 	QString userId = mAccountSettings->edtAccountId->text();
 	qulonglong uid = userId.toULongLong( &bOk );
-	
+
 	if( !bOk || uid == 0 || userId.isEmpty() )
 	{	KMessageBox::queuedMessageBox(this, KMessageBox::Sorry,
 	 	                              i18n("<qt>You must enter a valid ICQ No.</qt>"), i18n("ICQ"));
